include/nbl/ext/MitsubaLoader/CMitsubaLoader.h

```diff
@@ -31,27 +31,27 @@ class CMitsubaLoader : public asset::IRenderpassIndependentPipelineLoader
 		friend class CMitsubaMaterialCompilerFrontend;
 	public:
 		//! Constructor
-		CMitsubaLoader(asset::IAssetManager* _manager, io::IFileSystem* _fs);
+		CMitsubaLoader(asset::IAssetManager* _manager, system::ISystem* _system);
 
 		void initialize() override;
 
 	protected:
-		io::IFileSystem* m_filesystem;
+		system::ISystem* m_system;
```

Comment:
do not keep a handle to `ISystem` around, you don't need it (also dont take in the constructor

Reply:
if you need `m_system` to open new files, it should come from a Loader's context